src/EnergyPlus/WaterUse.cc

```diff
@@ -1092,117 +1099,142 @@ namespace WaterUse {
+                    if (!state.dataGlobal->WarmupFlag) {
+                        // print error for variables of hot water temperature
+                        ++this->CWHWTempErrorCount;
+                        TempDiff = this->ColdTemp - this->HotTemp;
+                        if (this->CWHWTempErrorCount < 2) {
+                            ShowWarningError(state,
+                                             format("CalcEquipmentFlowRates: \"{}\" - Hot water temperature is less than or equal to the cold water "
+                                                    "temperature ({:.2R} C)",
+                                                    this->Name,
+                                                    TempDiff));
+                            ShowContinueErrorTimeStamp(state, "");
+                            ShowContinueError(state, format("...hot water temperature        = {:.2R} C", this->HotTemp));
+                            ShowContinueError(state, format("...cold water temperature       = {:.2R} C", this->ColdTemp));
+                            ShowContinueError(state,
+                                              "...Hot water temperature should be greater than or equal to the cold water temperature. "
+                                              "Verify temperature setpoints and schedules.");
+                        } else {
+                            ShowRecurringWarningErrorAtEnd(
+                                state,
+                                format("\"{}\" - Hot water temperature should be greater than the cold water temperature error continues... ",
+                                       this->Name),
+                                this->CWHWTempErrIndex,
+                                TempDiff,
+                                TempDiff);
+                        }
                     }
+                } else {
+                    // HotMassFlowRate should always be between 0 and max
+                    this->HotMassFlowRate = this->TotalMassFlowRate * (this->TargetTemp - this->ColdTemp) / (this->HotTemp - this->ColdTemp);
                 }
             }
 
             this->ColdMassFlowRate = this->TotalMassFlowRate - this->HotMassFlowRate;
-
-            if (this->ColdMassFlowRate < 0.0) this->ColdMassFlowRate = 0.0;
-
             this->MixedTemp = (this->ColdMassFlowRate * this->ColdTemp + this->HotMassFlowRate * this->HotTemp) / this->TotalMassFlowRate;
+            // there should be no out of bounds results
+            assert(this->ColdMassFlowRate >= 0.0 && this->ColdMassFlowRate <= this->TotalMassFlowRate);
+            assert(this->HotMassFlowRate >= 0.0 && this->HotMassFlowRate <= this->TotalMassFlowRate);
+            assert(std::abs(this->HotMassFlowRate + this->ColdMassFlowRate - this->TotalMassFlowRate) < EPSILON);
         } else {
             this->HotMassFlowRate = 0.0;
-            this->ColdMassFlowRate = 0.0;
+            this->ColdMassFlowRate = this->TotalMassFlowRate;
```

Comment:
And if not MassFlow > 0 || this->allowHotControl, then ColdMassFlowRate should be = to what was calculated before entering this if block.

... and done.

Reply:
All cold here, :+1: